### PR TITLE
Refactor current models to match specs

### PR DESCRIPTION
**Pull request by hurrycane**
Adding users and contestants.

Contains rabl and oj examples.

### CODE REVIEW

**Comment by robertdolca** [comment]
&nbsp;<b>Code Climate</b> has <a href="https://codeclimate.com/repos/550c2ace69568065e600302d/compare/bogdan/refactor_current_models_to_match_specs">analyzed this pull request</a>.

**Comment by hurrycane** [comment]
@robertdolca See controller and rabl/oj views as an example. Also keep in mind that this will break current heroku setup.

You'll need to destroy current schema and run the new migrations.

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -1,25 +1,45 @@
-require 'factory_girl_rails'
-require 'codeclimate-test-reporter'
-
-CodeClimate::TestReporter.start
```
Comment (hurrycane) [inline spec/spec_helper.rb]: This should be kept.

@robertdolca Can you move this into the new spec_helper please?

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -6,6 +6,14 @@ class ApplicationController < ActionController::Base
   before_filter :cors_preflight_check
   after_filter :cors_set_access_control_headers
 
+  # always render json
+  before_action do |controller|
+    klass = controller.class.to_s
+    unless ["RailsAdmin", "Devise"].include?(klass.deconstantize)
```
Comment (palcu) [inline app/controllers/application_controller.rb]: Are you sure we don't want to do the Devise authentication using the frontend? I think it's easy to send the user and the password through JSON and get a JSON object for the user.

Comment (hurrycane) [inline app/controllers/application_controller.rb]: Check this out:
http://fancypixel.github.io/blog/2015/01/28/react-plus-flux-backed-by-rails-api/

**Comment by palcu** [comment]
LGTM from me.

**Comment by palcu** [comment]
I cannot create a user without `discourse_id`. I think this should not be a hard validation rule.

**Comment by robertdolca** [comment]
Heroku acum suporta automatic pull request deployment. Ca sa testezi poti intra aici infoeducatie-api-pr-35.herokuapp.com

**Pull request merged**